src/Semantic/Api/Terms.hs

```diff
@@ -106,27 +84,70 @@ quietTerm blob = showTiming blob <$> time' ( (doParse blob >>= withSomeTerm (fma
 
 type ParseEffects sig m = (Member (Error SomeException) sig, Member (Reader PerLanguageModes) sig, Member Parse sig, Member (Reader Config) sig, Carrier sig m)
 
-type TermConstraints =
- '[ Taggable
-  , Declarations1
-  , ConstructorName
-  , HasTextElement
-  , Show1
-  , ToJSONFields1
-  , Traversable
-  ]
-
-doParse :: (ParseEffects sig m) => Blob -> m (SomeTerm TermConstraints Loc)
-doParse blob = case blobLanguage blob of
-  Go         -> SomeTerm <$> parse goParser blob
-  Haskell    -> SomeTerm <$> parse haskellParser blob
-  JavaScript -> SomeTerm <$> parse tsxParser blob
-  JSON       -> SomeTerm <$> parse jsonParser blob
-  JSX        -> SomeTerm <$> parse tsxParser blob
-  Markdown   -> SomeTerm <$> parse markdownParser blob
-  Python     -> SomeTerm <$> parse pythonParser blob
-  Ruby       -> SomeTerm <$> parse rubyParser blob
-  TypeScript -> SomeTerm <$> parse typescriptParser blob
-  TSX        -> SomeTerm <$> parse tsxParser blob
-  PHP        -> SomeTerm <$> parse phpParser blob
+
+class ShowTerm term where
+  showTerm :: (Carrier sig m, Member (Reader Config) sig) => term Loc -> m Builder
```

Comment:
Same thought on `Show` here? `SerializeTerm` might be better?

Reply:
Again, it’s specifically supposed to use `Show`.